front/src/applications/editor/tools/routeEdition/components.tsx

```diff
@@ -34,8 +34,21 @@ export const RouteEditionLayers: FC = () => {
   } = useContext(EditorContext) as ExtendedEditorContextType<RouteEditionState>;
   const { mapStyle, layersSettings, issuesSettings } = useSelector(getMap);
 
-  const { routeState } = state as EditRoutePathState;
-  const selectionList = compact([routeState.entryPoint?.id, routeState.exitPoint?.id]);
+  const { routeState, optionsState } = state as EditRoutePathState;
+  const selectedRouteIndex =
+    optionsState.type === 'options' ? optionsState.focusedOptionIndex : undefined;
+  const selectedRouteDetectors =
```

Comment:
```suggestion
  const selectedRouteDetectors = selectedRouteIndex && optionsState.options ? optionsState.options[selectedRouteIndex].data.detectors : []
```
If you want to avoid using `!`